l1-contracts/test/foundry/l2/integration/L2Utils.sol

```diff
@@ -54,9 +68,61 @@ library L2Utils {
      * @dev Initializes the system contracts.
      * @dev It is a hack needed to make the tests be able to call system contracts directly.
      */
-    function initSystemContracts() internal {
+    function initSystemContracts(SystemContractsArgs memory _args) internal {
         bytes memory contractDeployerBytecode = readSystemContractsBytecode("ContractDeployer");
         vm.etch(DEPLOYER_SYSTEM_CONTRACT, contractDeployerBytecode);
+        forceDeploySystemContracts(_args);
+    }
+
+    function forceDeploySystemContracts(SystemContractsArgs memory _args) internal {
```

Comment:
the logic seems very similar to `function initSystemContracts`, why can't it be reused between the two places?

Reply:
Because one is for L1 and the other for L2

Reply:
We could use the L1 method for the L2 though... but the L2 uses forceDeployments, which is good to test anyway ( and it was here before I added the L1 method)

Reply:
Actually, we should just use the L2GatewayUpgradeHelper here